spec/models/goods_nomenclature_spec.rb

```diff
@@ -384,4 +384,180 @@
+    end
+
+    context 'when the goods nomenclature has no siblings' do
+      subject(:siblings) { create(:goods_nomenclature, :without_siblings).siblings }
+
+      it { expect(siblings).to be_empty }
+    end
+  end
+
+  describe '#children' do
+    context 'when the goods nomenclature has children' do
+      subject(:child_sids) { create(:goods_nomenclature, :with_children).children.map(&:goods_nomenclature_sid) }
+
+      it { is_expected.to eq([2]) }
+    end
+
+    context 'when the goods nomenclature has no children' do
+      subject(:child_sids) { create(:goods_nomenclature, :without_children).children.map(&:goods_nomenclature_sid) }
+
+      it { is_expected.to be_empty }
+    end
+  end
+
+  describe '#descendants' do
+    context 'when the goods nomenclature has descendants' do
+      subject(:descendant_sids) { create(:goods_nomenclature, :with_descendants).descendants.map(&:goods_nomenclature_sid) }
+
+      it { is_expected.to match_array([2, 3]) }
+    end
+
+    context 'when the goods nomenclature has no descendants' do
+      subject(:descendant_sids) { create(:goods_nomenclature, :without_descendants).descendants.map(&:goods_nomenclature_sid) }
+
+      it { is_expected.to be_empty }
+    end
+  end
+
+  describe '#heading?' do
+    context 'when the goods nomenclature has a heading goods nomenclature item id' do
+      subject(:goods_nomenclature) { create(:goods_nomenclature, :heading) }
+
+      it { is_expected.to be_heading }
+    end
+
+    context 'when the goods nomenclature has a non-heading goods nomenclature item id' do
+      subject(:goods_nomenclature) { create(:goods_nomenclature, :commodity) }
+
+      it { is_expected.not_to be_heading }
+    end
+  end
+
+  describe '#not_heading?' do
+    context 'when the goods nomenclature has a heading goods nomenclature item id' do
+      subject(:goods_nomenclature) { create(:goods_nomenclature, :heading) }
+
+      it { is_expected.not_to be_not_heading }
+    end
+
+    context 'when the goods nomenclature has a non-heading goods nomenclature item id' do
+      subject(:goods_nomenclature) { create(:goods_nomenclature, :commodity) }
+
+      it { is_expected.to be_not_heading }
+    end
+  end
+
+  describe '#chapter?' do
+    context 'when the goods nomenclature has a chapter goods nomenclature item id' do
+      subject(:goods_nomenclature) { create(:goods_nomenclature, :chapter) }
+
+      it { is_expected.to be_chapter }
+    end
+
+    context 'when the goods nomenclature has a non-chapter goods nomenclature item id' do
+      subject(:goods_nomenclature) { create(:goods_nomenclature, :commodity) }
+
+      it { is_expected.not_to be_chapter }
+    end
+  end
+
+  describe '#not_chapter?' do
+    context 'when the goods nomenclature has a chapter goods nomenclature item id' do
+      subject(:goods_nomenclature) { create(:goods_nomenclature, :chapter) }
+
+      it { is_expected.not_to be_not_chapter }
```

Comment:
is it expected to be a chapter? 😕

Reply:
This is confusing/unnecessary. Have removed